packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -158,7 +157,7 @@ export const usePickerValue = <
     onChange,
     value: inValueWithoutRenderTimezone,
     defaultValue: inDefaultValue,
-    closeOnSelect = variant === 'desktop',
+    closeOnSelect = false,
```

Comment:
You can remove `variant` from `UsePickerValueParams`

Reply:
Thanks for spotting this. 👍 
I had to shuffle some props around. 😉 
https://github.com/mui/mui-x/pull/15944/commits/ee93a4496792868a7454fd270fb1ae2cd0c6697a